File: src/main/java/net/insprill/cjm/hooks/VanishNoPacketHook.java
package net.insprill.cjm.hooks;

import net.insprill.cjm.handlers.VanishHandler;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.kitteh.vanish.event.VanishStatusChangeEvent;

public class VanishNoPacketHook implements Listener {

    @EventHandler(priority = EventPriority.MONITOR)
    public void onVanishToggle(VanishStatusChangeEvent e) {
        VanishHandler.handleVanishToggle(e.getPlayer(), e.isVanishing());
    }

}

File: src/main/kotlin/net/insprill/cjm/compatibility/hook/VanishNoPacketHook.kt
package net.insprill.cjm.compatibility.hook

import net.insprill.cjm.CustomJoinMessages
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.kitteh.vanish.event.VanishStatusChangeEvent

class VanishNoPacketHook(private val plugin: CustomJoinMessages) : Listener {

    @EventHandler(priority = EventPriority.MONITOR)
    fun onVanishToggle(e: VanishStatusChangeEvent) {
        plugin.vanish.handleToggle(e.player, e.isVanishing)
    }

}